src/common/components/Translator.tsx

```diff
@@ -522,6 +522,15 @@ function InnerTranslator(props: IInnerTranslatorProps) {
         highlightRef.current.handleInput()
     }, [selectedWord, highlightWords])
 
+    useEffect(() => {
+        if (selectedWord === '' || settings?.readSelectedWordsText === false) {
+            return
+        }
+        handleSelectedWordsSpeakAction()
+            .then()
+            .catch((reason) => console.error('read selected words error: ', reason))
+    }, [selectedWord])
```

Comment:
```suggestion
    }, [selectedWord, handleSelectedWordsSpeakAction, settings?.readSelectedWordsText])
```

Reply:
I tried adding `handleSelectedWordsSpeakAction` to the useEffect `deps` array and running it, but it keeps throwing errors. Ideally, we only need to trigger it when `selectedWord` changes, so adding `handleSelectedWordsSpeakAction` to deps might not be logically necessary. If my understanding is incorrect, please let me know. Thank you.

Reply:
The code has been refactored to use useCallback in commit 017f699484cac9dee5a9c8b57fbc2c6b6fbcdb84.